src/main/java/org/jenkinsci/plugins/docker/commons/credentials/DockerRegistryToken.java

```diff
@@ -96,6 +96,11 @@ public Void call() throws IOException {
                     if (config.exists()) {
                         json = JSONObject.fromObject(FileUtils.readFileToString(config, "UTF-8"));
                         auths = json.getJSONObject("auths");
+                        if (auths.isNullObject()) {
+                            auths = new JSONObject();
+                            json.put("auths", auths);
+                            FileUtils.writeStringToFile(config, json.toString(2), "UTF-8");
```

Comment:
I would add some logic on the FINE/CONFIG level here. It is just a data migration logic IIUC